src/Sylius/Bundle/CoreBundle/Test/Services/PaymentMethodFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Test\Services;
+
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class PaymentMethodFactory implements PaymentMethodFactoryInterface
+{
+    /**
+     * @var FactoryInterface
+     */
+    private $defaultFactory;
+
+    /**
+     * @var array
+     */
+    private $expectedGateways = [
+        'paypal_express_checkout' => 'paypal express checkout',
+        'be2bill_direct' => 'be2bill direct',
+        'be2bill_offsite' => 'be2bill offsite',
+        'stripe_checkout' => 'stripe checkout',
+        'offline' => 'offline',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(FactoryInterface $defaultFactory)
+    {
+        $this->defaultFactory = $defaultFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createNew()
+    {
+        return $this->defaultFactory->createNew();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createFromArray(array $parameters)
```

Comment:
If we see the need for such method in all resources, we could add this to default factory implementation, although I am not really sure. It is a bit "magic".